py-scripts/lf_interop_throughput.py

```diff
@@ -1589,48 +1774,104 @@ def generate_report(self, iterations_before_test_stopped_by_user, incremental_ca
+                        _obj="The below tables provides detailed information for the throughput test on each device.")
                 report.build_objective()
                 self.mac_id_list = [item.split()[-1] if ' ' in item else item for item in self.mac_id_list]
                 if self.expected_passfail_value or self.device_csv_name:
                     test_input_list, pass_fail_list = self.get_pass_fail_list(device_type, incremental_capacity_list[i], devices_on_running, download_data, upload_data)
-                bk_dataframe = {
-                    " Device Type ": device_type[0:int(incremental_capacity_list[i])],
-                    " Username": devices_on_running[0:int(incremental_capacity_list[i])],
-                    " SSID ": self.ssid_list[0:int(incremental_capacity_list[i])],
-                    " MAC ": self.mac_id_list[0:int(incremental_capacity_list[i])],
-                    " Channel ": self.channel_list[0:int(incremental_capacity_list[i])],
-                    " Mode": self.mode_list[0:int(incremental_capacity_list[i])],
-                    # " Direction":direction_in_table[0:int(incremental_capacity_list[i])],
-                    " Offered download rate ": download_list[0:int(incremental_capacity_list[i])],
-                    " Observed Average download rate ": [str(n) + " Mbps" for n in download_data[0:int(incremental_capacity_list[i])]],
-                    " Offered upload rate ": upload_list[0:int(incremental_capacity_list[i])],
-                    " Observed Average upload rate ": [str(n) + " Mbps" for n in upload_data[0:int(incremental_capacity_list[i])]],
-                    " RSSI ": ['' if n == 0 else '-' + str(n) + " dbm" for n in rssi_data[0:int(incremental_capacity_list[i])]],
-                    # " Link Speed ":self.link_speed_list[0:int(incremental_capacity_list[i])],
-                    " Packet Size(Bytes) ": [str(n) for n in packet_size_in_table[0:int(incremental_capacity_list[i])]]
-                }
-                if self.direction == "Bi-direction":
-                    bk_dataframe[" Average Rx Drop B% "] = upload_drop
-                    bk_dataframe[" Average Rx Drop A% "] = download_drop
-                elif self.direction == 'Download':
-                    bk_dataframe[" Average Rx Drop A% "] = download_drop
-                    # adding rx drop while uploading as 0
-                    bk_dataframe[" Average Rx Drop B% "] = [0.0] * len(download_drop)
-
+                if self.group_name:
+                    for key, val in self.group_device_map.items():
+                        if self.expected_passfail_value or self.device_csv_name:
+                            # Generating Dataframe when Groups with their profiles and pass_fail case is specified
+                            dataframe = self.generate_dataframe(val,
+                                                                device_type[0:int(incremental_capacity_list[i])],
```

Comment:
I'm not super supportive of this style of coding, but it was here before this PR and I don't care to look too closely. Fine as is.